rl4co/models/common/constructive/autoregressive/policy.py

```diff
@@ -0,0 +1,49 @@
+from typing import Union
+
+from rl4co.envs import RL4COEnvBase
+from rl4co.models.common.constructive.base import ConstructivePolicy
+
+from .decoder import AutoregressiveDecoder
+from .encoder import AutoregressiveEncoder
+
+
+class AutoregressivePolicy(ConstructivePolicy):
+    """Template class for an autoregressive policy, simple wrapper around
+    :class: rl4co.models.common.constructive.base.ConstructivePolicy.
+
+    Note:
+        While a decoder is required, an encoder is optional and will be initialized to
+        :class: rl4co.models.common.constructive.autoregressive.encoder.NoEncoder .
+        This can be used in decoder-only models in which at each step actions do not depend on
+        previously encoded states.
+    """
+
+    def __init__(
+        self,
+        encoder: AutoregressiveEncoder = None,
+        decoder: AutoregressiveDecoder = None,
```

Comment:
We shouldn't: the reason why it's like this is because I like to have "encoder" before "decoder" in order. In fact, encoding is not actually always necessary... So we can just flip the two I guess:
```python
    def __init__(
        self,
        decoder: AutoregressiveDecoder,
        encoder: AutoregressiveEncoder = None,
```